src/docs/development/platform-integration/c-interop.md

```diff
@@ -0,0 +1,342 @@
+---
+title: "Binding to native code using dart:ffi"
+description: "To use C code in your Flutter program, use the dart:ffi library (currently in preview)."
+---
+
+Flutter mobile can use the [dart:ffi][] library
+to call native C APIs. The _ffi_ acronym stands for
+[_foreign function interface._][FFI]
+Other terms for similar functionality include
+_native interface_ and _language bindings._
+
+This tutorial demonstrates how to bundle C/C++
+sources in a Flutter plugin and bind to them using
+the Dart FFI library on both Android and iOS.
+In this walkthrough, you'll create a C function
+that implements 32-bit addition and then
+exposes it through a Dart plugin named "native_add".
+
+{{ site.alert.note }}
+  The dart:ffi library is [in active development][ffi issue]
+  and isn't complete yet. Note that the API is likely to have
+  breaking changes between now and its completion.
+
+  Using the feature requires a Flutter 1.10.x
+  dev channel build. To switch to the dev channel and
+  upload the latest dev version, do the following:
+
+  ```terminal
+  $ flutter channel dev
+  $ flutter upgrade
+  ```
+  For more information on Flutter's channels,
+  see [Upgrading Flutter][].
+{{ site.alert.end }}
+
+API documentation is available from the Dart dev channel:
+[Dart API reference documentation][].
+
+## Step 1: Create a plugin
+
+If you already have a plugin, skip this step.
+
+To create a plugin called "native_add",
+do the following:
+
+```dart
+$ flutter create --template=plugin native_add
+$ cd native_add
+```
+
+## Step 2: Add C/C++ sources
+
+You need to inform both the Android and iOS build
+systems about the native code so the code can be compiled
+and linked appropriately into the final application.
+
+You add the sources to the `ios` folder,
+because CocoaPods doesn't allow including sources
+above the podspec file, but Gradle allows you to point
+to the `ios` folder. It's not required to use the same
+sources for both iOS and Android;
+you may, of course, add Android-specific sources
+to the `android` folder and modify the `CMakeLists.txt`
+file appropriately.
+
+The FFI library can only bind against C symbols,
+so in C++ these symbols must be marked "extern C".
+You should also add attributes to indicate that the
+symbols are referenced from Dart,
+to prevent a linker from discarding the symbols
+during link-time optimization.
+
+For example:
+
+```
```

Comment:
This quote is Bash, a command-line which can be run from the root of the directly. Can we make it show up with bash highlighting in the website?